compiler/src/dotty/tools/dotc/semanticdb/Tools.scala

```diff
@@ -85,42 +87,8 @@ object Tools:
     case UNKNOWN_LANGUAGE | Unrecognized(_) => "unknown"
   end languageString
 
-  private def processSymbol(info: SymbolInformation)(using sb: StringBuilder): Unit =
-    import SymbolInformation.Kind._
-    sb.append(info.symbol).append(" => ")
-    if info.isAbstract then sb.append("abstract ")
-    if info.isFinal then sb.append("final ")
-    if info.isSealed then sb.append("sealed ")
-    if info.isImplicit then sb.append("implicit ")
-    if info.isLazy then sb.append("lazy ")
-    if info.isCase then sb.append("case ")
-    if info.isCovariant then sb.append("covariant ")
-    if info.isContravariant then sb.append("contravariant ")
-    if info.isVal then sb.append("val ")
-    if info.isVar then sb.append("var ")
-    if info.isStatic then sb.append("static ")
-    if info.isPrimary then sb.append("primary ")
-    if info.isEnum then sb.append("enum ")
-    if info.isDefault then sb.append("default ")
-    info.kind match
-      case LOCAL => sb.append("local ")
-      case FIELD => sb.append("field ")
-      case METHOD => sb.append("method ")
-      case CONSTRUCTOR => sb.append("ctor ")
-      case MACRO => sb.append("macro ")
-      case TYPE => sb.append("type ")
-      case PARAMETER => sb.append("param ")
-      case SELF_PARAMETER => sb.append("selfparam ")
-      case TYPE_PARAMETER => sb.append("typeparam ")
-      case OBJECT => sb.append("object ")
-      case PACKAGE => sb.append("package ")
-      case PACKAGE_OBJECT => sb.append("package object ")
-      case CLASS => sb.append("class ")
-      case TRAIT => sb.append("trait ")
-      case INTERFACE => sb.append("interface ")
-      case UNKNOWN_KIND | Unrecognized(_) => sb.append("unknown ")
-    sb.append(info.displayName).nl
-  end processSymbol
+  private def processSymbol(info: SymbolInformation, printer: SymbolInfomationPrinter)(using sb: StringBuilder): Unit =
+    sb.append(printer.pprintSymbolInformation(info)).nl
```

Comment:
Should we much priority on string concatenation efficiency?
- I personally prefer returning String instead of passing `StringBuilder` (because of its code readability).
- While I understand `StringBuilder` concatenation is generally faster, this pretty printer is used only for testing and the efficiency might be not a big deal?
- btw, what do you think about this approach? https://github.com/lampepfl/dotty/pull/12885#discussion_r658769494

Reply:
Its true the efficiency is not the priority before the correctness of the actual output, so this would only be the last thing to address.

I have addressed that other comment now.